google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/GoogleApiClientAppEngineAdminService.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.application;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.appengine.v1.Appengine.Apps;
+import com.google.api.services.appengine.v1.model.Application;
+import com.google.api.services.appengine.v1.model.ListLocationsResponse;
+import com.google.api.services.appengine.v1.model.Location;
+import com.google.api.services.appengine.v1.model.Operation;
+import com.google.api.services.appengine.v1.model.Status;
+import com.google.cloud.tools.intellij.resources.GoogleApiClientFactory;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * An implementation of {@link AppEngineAdminService} that uses a Google API Client to communicate
+ * with the App Engine service.
+ */
+public class GoogleApiClientAppEngineAdminService extends AppEngineAdminService {
+
+  private final static String APP_ENGINE_RESOURCE_WILDCARD = "-";
+  private final long CREATE_APPLICATION_POLLING_INTERVAL_MS = 1000;
```

Comment:
Good catch. I'm surprised our static analysis tools didn't pick this up. I seem to remember this sort of thing failing CI builds in the past...